File: frontend/bff/src/application/policies/commands/create-policy.command.ts
import type { RequestContext } from '../../../cross-cutting/context/request-context.js';

export interface CreatePolicyCommand {
  context: RequestContext;
  code: string;
  title: string;
  version: string;
  description: string | null;
  validFrom: string;
  validTo: string | null;
  /** Inicia com os eixos, instrumentos e bandas do modelo FIX. */
  useTemplate: boolean;
}

File: backend/core-service/src/Fix.Application/UseCases/Policies/Commands/CreatePolicyCommand.cs
using Fix.Application.Abstractions.Authorization;
using Fix.Application.Abstractions.Context;
using Fix.Application.Abstractions.Messaging;
using Fix.Application.Policies.Dtos;
using Fix.Domain.AggregateRoots.Roles;

namespace Fix.Application.Policies.Commands;

[RequireRole(RoleCodes.CreatePolicy)]
public sealed record CreatePolicyCommand(
    Guid UserId,
    Guid OrganizationId,
    string Code,
    string Title,
    string Version,
    string? Description,
    DateOnly ValidFrom,
    DateOnly? ValidTo,
    bool UseTemplate)
    : ICommand<PolicyDto>, IOrganizationRequest;

